### ISSUE TITLE
undefined mmsghdr when building for netbsd/arm

### ISSUE THREAD

**Issue by JeremyRand** [miekg/dns#655]
# Steps to reproduce

~~~
env GOOS=netbsd GOARCH=arm go get github.com/miekg/dns/...
~~~

# Expected results

It should build without errors.

# Observed results

~~~
# github.com/miekg/dns/vendor/golang.org/x/net/internal/socket
src/github.com/miekg/dns/vendor/golang.org/x/net/internal/socket/mmsghdr_unix.go:11:17: undefined: mmsghdr
~~~

# Other notes

Seems to be caused by a vendored library not being updated to include the fix to https://github.com/golang/go/issues/24346 .

**Comment by miekg** [comment]
Please send PR to update the vendored stuff.

**Comment by andrewtj** [comment]
Is this fixed after the vendor update? (c9cd01b)

**Comment by JeremyRand** [comment]
Yes, seems to be fixed now.  Thanks.
